src/provider.ts

```diff
@@ -94,10 +94,19 @@ export default class VSCodeSearchProvider implements AppSearchProvider {
 
   activateResult(result: string): void {
     if (this.app) {
-      this.app?.app_info.launch(
-        [Gio.file_new_for_path(this.workspaces[result].path)],
-        null,
-      );
+      const path = this.workspaces[result].path;
+      if (path.startsWith("vscode-remote://")) {
+        const lastSegment = path.split('/').pop();
+        const type = lastSegment?.includes('.') ? 'file' : 'folder';
```

Comment:
I think this check could be problematic, specially on linux where the `.` is used to hide files and folders, if the user open `.ssh` or `.config` or a personal folder like my [dotfiles](https://github.com/MrMarble/dotfiles) that are in a `.dotfiles` folder.  

Since its a remote path we can't check the type but we could ignore the first character to reduce false positives
```suggestion
        const type = lastSegment?.slice(1)?.includes('.') ? 'file' : 'folder';
```

Reply:
Perfect